src/client/app/actions/admin.ts

```diff
@@ -46,45 +46,37 @@ function markPreferencesSubmitted(): t.MarkPreferencesSubmittedAction {
 
 
 function fetchPreferences(): Thunk {
-	return (dispatch: Dispatch, getState: GetState) => {
+	return async (dispatch: Dispatch, getState: GetState) => {
 		dispatch(requestPreferences());
-		return axios.get('/api/preferences')
-			.then(response => {
-				dispatch(receivePreferences(response.data));
-				if (!getState().graph.hotlinked) {
-					dispatch((dispatch2: Dispatch, getState2: GetState) => {
-						const state = getState();
-						dispatch2(changeChartToRender(state.admin.defaultChartToRender));
-						if (response.data.defaultBarStacking !== state.graph.barStacking) {
-							dispatch2(changeBarStacking());
-						}
-					});
+		const preferences = await preferencesApi.getPreferences();
```

Comment:
I know the old code did not have a catch condition, but should we be explicit and add a catch to every request?

Reply:
Maybe? I feel like we want to throw an error here, at least until we have more robust error handling (which isn't in scope for this pr, I don't think).